presto-main/src/main/java/com/facebook/presto/eventlistener/EventListenerManager.java

```diff
@@ -59,16 +59,24 @@ public void loadConfiguredEventListener()
             throws Exception
     {
         if (EVENT_LISTENER_CONFIGURATION.exists()) {
-            Map<String, String> properties = new HashMap<>(loadProperties(EVENT_LISTENER_CONFIGURATION));
-
-            String eventListenerName = properties.remove(EVENT_LISTENER_PROPERTY_NAME);
-            checkArgument(!isNullOrEmpty(eventListenerName),
-                    "Access control configuration %s does not contain %s", EVENT_LISTENER_CONFIGURATION.getAbsoluteFile(), EVENT_LISTENER_PROPERTY_NAME);
-
-            setConfiguredEventListener(eventListenerName, properties);
+            Map<String, String> properties = loadProperties(EVENT_LISTENER_CONFIGURATION);
+            checkArgument(
+                    !isNullOrEmpty(properties.get(EVENT_LISTENER_PROPERTY_NAME)),
+                    "Access control configuration %s does not contain %s",
+                    EVENT_LISTENER_CONFIGURATION.getAbsoluteFile(),
+                    EVENT_LISTENER_PROPERTY_NAME);
+            loadConfiguredEventListener(properties);
         }
     }
 
+    public void loadConfiguredEventListener(Map<String, String> properties)
+    {
+        properties = new HashMap<>(properties);
+        String eventListenerName = properties.remove(EVENT_LISTENER_PROPERTY_NAME);
```

Comment:
Why not simply `get(EVENT_LISTENER_PROPERTY_NAME)` ?

Reply:
This is per convention. One of the properties defines what event listener provider to use, the other properties is the actual configuration that must be passed to the provider. The property that defines what provider to use must be removed, as the provider itself doesn't understand it.